analyzers/tests/SonarAnalyzer.Test/Helpers/DisjointSetsPrimitivesTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿using static SonarAnalyzer.Helpers.DisjointSetsPrimitives;
+
+namespace SonarAnalyzer.Test.Helpers;
+
+[TestClass]
+public class DisjointSetsPrimitivesTest
+{
+    [TestMethod]
+    public void FindRootAndUnion_AreConsistent()
+    {
+        var parents = Enumerable.Range(1, 5).ToDictionary(i => i.ToString(), i => i.ToString());
+        foreach (var item in parents.Keys)
+        {
+            FindRoot(parents, item).Should().Be(item);
+        }
+
+        Union(parents, "1", "1");
+        FindRoot(parents, "1").Should().Be("1");                    // Reflexivity
+        Union(parents, "1", "2");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Correctness
+        Union(parents, "1", "2");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Idempotency
+        Union(parents, "2", "1");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Symmetry
+
+        FindRoot(parents, "3").Should().Be("3");
+        Union(parents, "2", "3");
+        FindRoot(parents, "2").Should().Be(FindRoot(parents, "3"));
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "3")); // Transitivity
+        Union(parents, "3", "4");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "4")); // Double transitivity
+        Union(parents, "4", "1");
+        FindRoot(parents, "4").Should().Be(FindRoot(parents, "1")); // Idempotency after transitivity
+    }
+
+    [TestMethod]
+    public void DisjointSetsAndUnion_AreConsistent()
+    {
+        var parents = Enumerable.Range(1, 5).ToDictionary(i => i.ToString(), i => i.ToString());
+        DisjointSets(parents).Should().BeEquivalentTo<List<string>>([["1"], ["2"], ["3"], ["4"], ["5"]]); // Initial state
+        Union(parents, "1", "2");
+        DisjointSets(parents).Should().BeEquivalentTo<List<string>>([["1", "2"], ["3"], ["4"], ["5"]]);   // Correctness
+        Union(parents, "1", "2");
+        DisjointSets(parents).Should().BeEquivalentTo<List<string>>([["1", "2"], ["3"], ["4"], ["5"]]);   // Idempotency
+
+        Union(parents, "2", "3");
+        DisjointSets(parents).Should().BeEquivalentTo<List<string>>([["1", "2", "3"], ["4"], ["5"]]);
```

Comment:
Isn't this indirect `transitivity`? Did you forget to add a comment or do you think it's not worth mentioning it here?

Reply:
The goal of the test was to show idempotency after transitivity, and I needed to assert the same expected value twice to show the idempotency. Anyway, it doesn't hurt to also mention that the first step takes advantage of the transitivity, so I have added the comment, as you suggest.